src/notificationarea.cpp

```diff
@@ -131,6 +169,20 @@ void NotificationArea::setSettings(const QString &placement, int width, int spac
     m_spacing = spacing;
     m_layout->setSizes(m_spacing, width);
 
+    m_screenWithMouse = screenWithMouse;
+
+    if (m_screenWithMouse)
+    {
+        connect(qApp, &QGuiApplication::screenAdded, this, &NotificationArea::screenAdded, Qt::UniqueConnection);
+        connect(qApp, &QGuiApplication::screenRemoved, this, &NotificationArea::screenRemoved, Qt::UniqueConnection);
+
+        const auto screens = qApp->screens();
+        for (const auto &screen: screens)
+            connect(screen, &QScreen::availableGeometryChanged, this, &NotificationArea::availableGeometryChanged, Qt::UniqueConnection);
+    }
+    else
+        connect(qApp->primaryScreen(), &QScreen::availableGeometryChanged, this, &NotificationArea::availableGeometryChanged, Qt::UniqueConnection);
+
```

Comment:
When the value of `m_screenWithMouse` is changed by the user, we'll have redundant connections; won't we? That could cause trouble later.

Also, why `Qt::UniqueConnection` when we can check whether `m_screenWithMouse` is changed or not.

Reply:
> When the value of `m_screenWithMouse` is changed by the user, we'll have redundant connections; won't we? That could cause trouble later.

It shouldn't cause any trouble because those signals are used only when `m_screenWithMouse` is true but doesn't make any difference when `m_screenWithMouse` is false but you're right we should avoid it. I'll change it asap.